message.go

```diff
@@ -237,13 +238,18 @@ func (r *messageSetReaderV1) readMessage(min int64,
 			}
 
 			// read and decompress the contained message set.
-			var decompressed []byte
+			var decompressed bytes.Buffer
+
 			if r.remain, err = readBytesWith(r.reader, r.remain, func(r *bufio.Reader, sz, n int) (remain int, err error) {
-				var value []byte
-				if value, remain, err = readNewBytes(r, sz, n); err != nil {
-					return
-				}
-				decompressed, err = codec.Decode(value)
+				decompressed.Grow(4 * n)
```

Comment:
How did you arrive at the `4`?

Reply:
Intuition :D 

Efficient compression algorithms will compressed down to ~75%, so that should fit in the buffer without reallocating it too much.

Eventually we could reuse these buffers with a sync.Pool and let natural usage of the program grow the buffer size to whatever will fit.

Reply:
Fair enough.  Drop that as a comment for the next person? 😄